lib/Backend/arm64/ARM64Encoder.h

```diff
@@ -150,6 +150,8 @@ enum
     NEONREG_INVALID = 99,
 };
 
+#define LAST_FLOAT_REG_ENCODE NEONREG_D31
+
```

Comment:
Since you trimmed off a few FP registers, we might consider loading them with commonly-used constant values (0, 0.5) as a future optimization.